codec/decode.go

```diff
@@ -2202,6 +2225,11 @@ func setZero(iv interface{}) {
 }
 
 func (d *Decoder) decode(iv interface{}) {
+	d.remainingDepth--
+	if d.remainingDepth < 0 {
+		panic("max depth exceeded")
```

Comment:
Does this crash the program if it hits a malicious input?  how is this better than the status quo?

Reply:
No, `Decode` does error handling by catching panics and returning the panic value as an error.

Reply:
Ah ok, great.